engine/runtime/src/main/java/org/enso/interpreter/runtime/TruffleCompilerContext.java

```diff
@@ -723,8 +734,7 @@ public void close() {
         module.module.setLoadedFromCache(loadedFromCache);
       }
       if (resetScope) {
-        module.module.ensureScopeExists();
-        module.module.getScope().reset();
+        module.module.resetScope();
```

Comment:
I would expect this to call `newBuilder()`.

Reply:
`resetScope` does call `newBuilder`